server/src/main/java/org/cloudfoundry/identity/uaa/web/UaaLogFilter.java

```diff
@@ -0,0 +1,53 @@
+package org.cloudfoundry.identity.uaa.web;
+
+import com.ge.predix.log.filter.LogFilter;
+
+import org.slf4j.MDC;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.UUID;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UaaLogFilter extends LogFilter {
+    static final String CORRELATION_HEADER_NAME = "X-B3-TraceId";
+    static final String CORRELATION_KEY_NAME = "traceId";
+
+    public UaaLogFilter(final LinkedHashSet<String> hostnames, final LinkedHashSet<String> zoneHeaders,
+                     final String defaultZone) {
+        super(hostnames, zoneHeaders, defaultZone);
+    }
+
+    public UaaLogFilter() {
+        super();
+    }
+
+    @Override
+    protected void doFilterInternal(final HttpServletRequest request,
+                                    final HttpServletResponse response, final FilterChain filterChain)
+        throws ServletException, IOException {
+        try {
+            setCorrelationId(request, response);
+            super.doFilterInternal(request, response, filterChain);
+        } finally {
+            clearMDC();
+        }
+    }
+
+    private void setCorrelationId(final HttpServletRequest request, final HttpServletResponse response) {
+        String correlationId = request.getHeader(CORRELATION_HEADER_NAME);
+        if (StringUtils.isEmpty(correlationId)) {
+            correlationId = UUID.randomUUID().toString();
```

Comment:
This does not match the expected trace-id format. Check the trace-ids created by CF/AVI.